### ISSUE TITLE
How to properly handle the illegal output content of ChatGPT?

### ISSUE THREAD

**Issue by YuboFeng2023** [ArrogantL/ChatGPT4CausalReasoning#1]
Hi,

Thank you for sharing the code; it will encourage the development of ECI research!

I noticed that you [set a default value (0) for the prediction](https://github.com/ArrogantL/ChatGPT4CausalReasoning/blob/62eb315358df977cfcc1c0c00ee4c2d9d3ea56a8/ECI.py#L136).

This means that if ChatGPT outputs illegal content, such as "Unknown," you still determine that there is no causal relationship in this sentence.

Is this setting reasonable?

**Comment by ArrogantL** [comment]
How the responses of ChatGPT were mapped to the desired format?
Our goal is to reflect the actual performance of ChatGPT, avoiding inaccurate evaluation results stemming from wrong automated mapping rules.

For the CEG task:

no further text map is necessary; we directly use the responses of ChatGPT as its final prediction.

For the ECI task:

Under the zero-shot setting, we found that out of 138,166 instances, only 96 responses do not begin with "yes" or "no". Therefore, responses starting with "yes" will be assigned the label 1, while other cases will be assigned the label 0.

Under the ICL and CoT settings, we construct demonstrations ending with the labels "yes" or "no". Therefore, responses from ChatGPT ending with "yes" will be mapped to label 1, while others will be mapped to label 0.

For the prompts discussed in Section 5.3, these prompts are directional. Therefore, for a pair of events, we reverse their order to obtain two responses from ChatGPT. If at least one of the two responses starts with "yes", it is assigned the label 1, while other cases will be categorized as label 0.

For the open-ended prompt discussed in Section 5.7, we added descriptions in the prompt about the output format, allowing for easy parsing into causal event pairs.

For the CD task:

Most responses begin with phrases "option1", "option2", "yes", or "no", which can be easily mapped to labels. For other response formats, we manually assigned labels for each different format of response, and the mapping results are recorded in the supplementary file "CD_and_CEG_compute_score.py".
